utils/s2n_safety.h

```diff
@@ -111,3 +111,17 @@ extern int s2n_mul_overflow(uint32_t a, uint32_t b, uint32_t* out);
 extern int s2n_align_to(uint32_t initial, uint32_t alignment, uint32_t* out);
 extern int s2n_add_overflow(uint32_t a, uint32_t b, uint32_t* out);
 extern int s2n_sub_overflow(uint32_t a, uint32_t b, uint32_t* out);
+
+/**
+ * Define a C99-compliant static assert.
+ *
+ * Static_assert() is not defined before C11. So, we can't rely on it when we
+ * want to be C99 compliant. Instead, use a method that is guaranteed to be C99
+ * compliant and still give us an equivalent static assert mechanism.
+ *
+ * Taken from aws-c-common: https://github.com/awslabs/aws-c-common/blob/main/include/aws/common/macros.h#L19
+ */
+#define S2N_CONCAT(A, B) A##B
```

Comment:
Does this really need to be a macro?

Reply:
Yeah, to resolve/build the macro correctly.
E.g. this is using `##` straightforward: https://godbolt.org/z/aYPzz1os6
`__LINE__` is not pre-processed correctly.

Maybe there is another way without. But I know of this way.